### ISSUE TITLE
Cannot read property 'render' of undefined

### ISSUE THREAD

**Issue by thomas-jeepe** [mobxjs/mobx-react#63]
So I was switching my project to typescript and in this process I got  `Cannot read property 'render' of undefined`.

After following the stack trace, I ended up in this [line.](https://github.com/mobxjs/mobx-react/blob/master/index.js#L151)

Debugging showed that `componentClass.prototype` was undefined, causing it to throw on trying to get a property on undefined. Changing the line to the below fixed everything and all components worked.

```javascript
if (typeof componentClass === "function" && !componentClass.prototype && !componentClass.isReactClass && !React.Component.isPrototypeOf(componentClass)) {
```

[This is the stack trace, tsconfig, and relevant files](https://gist.github.com/thomas-jeepe/327dd0f86f30d67efe4da413e5063d77)

Im not sure if this is a bug on my end or on mobx-react. Right now the "fix" I am using is replacing the line of code, but that isn't very maintainable.

**Comment by mweststrate** [comment]
I just released `mobx-react@3.3.1`, could you verify whether this solves the issue? Spotted a potential error in the check indeed.

**Comment by thomas-jeepe** [comment]
Just upgraded to 3.3.1 and it fixed it thanks a ton!
